flink-core/src/test/java/org/apache/flink/util/FileUtilsTest.java

```diff
@@ -202,6 +202,24 @@ public void testDeleteDirectoryWhichIsAFile() throws Exception {
 		}
 	}
 
+	@Test
+	public void testDeleteSymbolicLinkDirectory() throws Exception {
+
+		// deleting a symbolic link directory should not delete the files in it.
+
+		File symbolicLinkDirectory = tmp.newFolder();
+		File symbolicLinkDirectoryChild = new File(symbolicLinkDirectory, "child");
+		//noinspection ResultOfMethodCallIgnored
+		symbolicLinkDirectoryChild.createNewFile();
+		File symbolicLink = tmp.newFile();
```

Comment:
Do we need to create and delete the file at all? It could just be `File symbolicLink = new File(tmp.getRoot(), "symLink");`